### PR TITLE
feat(environments): Show and hide environments

### PR DESCRIPTION
**Pull request by lynnagara**
Depends on https://github.com/getsentry/sentry/pull/7181

### CODE REVIEW

**Comment by MaxBittker** [comment]
there's a merge conflict with the snap

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/views/projectEnvironments.jsx [hunk 1]
```diff
@@ -9,30 +10,142 @@ import PanelBody from './settings/components/panelBody';
...
+  componentWillReceiveProps(nextProps) {
+    const isHidden = nextProps.route.path === 'environments/hidden/';
+    const environments = isHidden
+      ? EnvironmentStore.getHidden()
+      : EnvironmentStore.getActive();
+    this.setState(
+      {
+        isHidden,
+        environments,
+        title: isHidden ? t('Hidden') : t('Active Environments'),
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/views/projectEnvironments.jsx]: imo title should be derived in render and not put into state (nit)

File: tests/js/spec/views/projectEnvironments.spec.jsx [hunk 2]
```diff
@@ -6,44 +6,113 @@ import ProjectEnvironments from 'app/views/projectEnvironments';
...
+        url: baseUrl,
+      });
+    });
+    it('hides', function() {
+      EnvironmentStore.loadInitialData(TestStubs.Environments(false));
+      const wrapper = mountComponent(false);
+      wrapper
+        .find('Button')
+        .first()
+        .simulate('click');
```
Comment (MaxBittker) [inline tests/js/spec/views/projectEnvironments.spec.jsx]: nice tests

File: tests/js/spec/views/projectEnvironments.spec.jsx [hunk 3]
```diff
@@ -6,44 +6,113 @@ import ProjectEnvironments from 'app/views/projectEnvironments';
...
+  const project = TestStubs.Project();
+  let path = isHidden ? 'environments/hidden/' : 'environments/';
+  return mount(
+    <ProjectEnvironments
+      params={{
+        orgId: org.slug,
+        projectId: project.slug,
+      }}
+      route={{path}}
+    />,
```
Comment (MaxBittker) [inline tests/js/spec/views/projectEnvironments.spec.jsx]: https://github.com/getsentry/sentry/blob/getsentry-show-hide-environments/tests/js/setup.js#L44-L55

routerContext might be what you want here instead, it's less verbose for this common argument

File: src/sentry/static/sentry/app/views/projectSettings/index.jsx [hunk 4]
```diff
@@ -104,7 +104,10 @@ const ProjectSettings = createReactClass({
             >
               {t('Alerts')}
             </ListLink>
-            <ListLink to={`/${orgId}/${projectId}/settings/environments/`}>
+            <ListLink
+              to={`/${orgId}/${projectId}/settings/environments/`}
+              isActive={loc => path.indexOf(loc.pathname) === 0}
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/views/projectSettings/index.jsx]: would startsWith work here? its a bit more explicit

File: src/sentry/static/sentry/app/views/projectEnvironments.jsx [hunk 5]
```diff
@@ -43,11 +156,17 @@ const ProjectEnvironments = createReactClass({
...
+                index={true}
+              >
+                {t('Hidden')}
+              </ListLink>
             </ul>
           }
         />
         <Panel>
-          <PanelHeader>{t('Environments')}</PanelHeader>
+          <PanelHeader>{this.state.title}</PanelHeader>
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/views/projectEnvironments.jsx]: might as well take this out of state on line 143, but also i think it's better to do your ternary to decide title in the render function and not have this on state at all

Comment (MaxBittker) [reply src/sentry/static/sentry/app/views/projectEnvironments.jsx]: (one benefit is that if isHidden is updated, title will always match it no matter what)

**Review by MaxBittker — APPROVED** [review]
tried it out and it works nicely, code mostly lgtm, i left some small nits 👍

**Comment by MaxBittker** [comment]
screenshot 2018-02-12 17 28 04
screenshot 2018-02-12 17 27 57
screenshot 2018-02-12 17 27 51

File: src/sentry/static/sentry/app/views/projectSettings/index.jsx (hunk 4 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/projectSettings/index.jsx]: Yes, but I'd rather keep the implementation the same as everything else in this page

File: src/sentry/static/sentry/app/views/projectEnvironments.jsx (hunk 5 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/projectEnvironments.jsx]: Good point 👍

File: tests/js/spec/views/projectEnvironments.spec.jsx (hunk 3 above)
Comment (lynnagara) [reply tests/js/spec/views/projectEnvironments.spec.jsx]: Thanks! Forgot that existed

**Pull request merged**